evm/evm-codec/src/codecs/primitives.ts

```diff
@@ -14,119 +17,119 @@ export const bool: Codec<boolean> = {
   isDynamic: false,
 }
 
-export const uint8: Codec<number> = {
-  encode(sink: Sink, val: number) {
-    sink.u8(val)
+export const uint8: Codec<Numberish, number> = {
+  encode(sink: Sink, val: Numberish) {
+    sink.u8(safeToNumber(val, 0, 255))
```

Comment:
@eldargab can we just check here if it is safeInteger and sink (view) will de the rest?

Reply:
just to reduce overhead

Reply:
Yes, `DataView` set methods already check input boundaries.

Reply:
As turned out it doesn't